### PR TITLE
Introduce WT A11y Proposal 2023

### PR DESCRIPTION
**Pull request by carlos-zamora**
The release of Windows Terminal has served as a way to reinvigorate the command-line ecosystem on Windows by providing a modern experience that is consistently updated. This experience has caused faster innovation in the Windows command-line ecosystem that can be seen across various areas like expanded virtual terminal sequence support and enhanced accessibility. Command-line apps can now leverage these innovations to create a better experience for the end-user.

Since accessibility is a very broad area, this document is intended to present recent innovations in the accessibility space for the command-line ecosystem. Furthermore, it will propose additional improvements that can be made alongside key stakeholders that could benefit from such changes.

### CODE REVIEW

**Comment by carlos-zamora** [comment]
### Note to reader
If there are any features that would have a positive accessibility impact, mention them here. Accessibility comes in many different ways, so features outside of UI Automation deserve to be mentioned too.

Of course, we can add to this document as we come up with more ideas later too. 😉

File: doc/terminal-a11y-2023.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+This is tracked by [A11y: windows terminal emits selection/deselection events in mark mode when navigating with arrow keys · Issue #13447](https://github.com/microsoft/terminal/issues/13447).
+
+### Search improvements
+The search dialog can be used to perform a runtime query of the text buffer. However, in its current form, it does not count the total number of matches like other developer tools (i.e. Visual Studio Code). This work item tracks two things:
+1. Support for counting the total number of matches
+2. Displaying the number of results in the search box and announcing it via a UIA notification to the screen reader
+Additional search features may fall into this including but not limited to:
+- Highlight the search results in the scroll bar
+- Successful searches should read the line where text was found (similar to VS Code experience)
```
Comment (lhecker) [inline doc/terminal-a11y-2023.md:125]: FYI/BTW I benchmarked this and the runtime of `<regex>` practically depends entirely on the number of matches. Given a few thousand matches it takes about 100ms on a slower CPU, which is too much, but not _/too/_ much. `std::boyer_moore_horspool_searcher` would probably be a good alternative by default (>20x faster) if we consider that most of the time people aren't using the regexp search functionality in VS Code either.

**Review by lhecker — APPROVED** [review]

File: doc/terminal-a11y-2023.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+### Verification Tools
+[Accessibility Insights](https://accessibilityinsights.io/) is a modern tool that can be used to test and understand accessibility issues on Windows. It comes with a built-in color contrast analyzer, a UI Automation event listener, and a UI Automation tree explorer. Additionally, it can be used to explore and interact with different control patterns using the same API screen readers and other accessibility tools rely on. Accessibility Insights is also capable of running automated tests in CI and locally to detect common and simple issues.
+
+## Proposed Accessibility Improvements
+
+### Respect Text Size OS setting
+Windows Terminal has a profile setting for the font size (`"font"."size": <integer>`) which is then used when instantiating new terminal sessions. There should be a way to make the terminal font size respect the text size setting in Windows found in the Settings App > Accessibility > Text Size.
+
+A possible solution is to scale the session's text size at runtime based on the OS setting. Furthermore, the Windows Terminal's settings UI should warn the user when they are attempting to modify the font size profile setting.
```
[the remark this answers is not in the document]
Comment (DHowett) [reply doc/terminal-a11y-2023.md:67]: I think if the user is using this accessibility feature, they can be trusted to understand that "text scaling" means "text scaling wherever there's text."

We can display a hint in the settings UI if it's enabled:

image

File: doc/terminal-a11y-2023.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+### Internal Partners
+There are many first-party command-line applications on Windows. The following are a few examples of those that are regularly updated:
+- [**GitHub CLI**](https://cli.github.com/): a tool that can be used to query and interact with GitHub repos (open source)
+- [**Winget**](https://github.com/microsoft/winget-cli): a tool to install applications and other packages
+- [**PSReadLine**](https://github.com/PowerShell/PSReadLine): a PowerShell module that enhances the input line experience
+- [**Windows Subsystem for Linux (WSL)**](https://learn.microsoft.com/en-us/windows/wsl/): a tool to manage and run GNU/Linux environments without a traditional virtual machine
+- [**PowerShell**](https://github.com/PowerShell/PowerShell): a cross-platform command-line shell (open source)
+- **CMD**: a Windows-specific command-line shell
```
[the remark this answers is not in the document]
Comment (carlos-zamora) [reply doc/terminal-a11y-2023.md:20]: Whoops, yeah that's not the case haha. My bad.

**Review by zadjii-msft — APPROVED** [review]
Yea I don't have much to add to this. This is all good. The minutia of the VT sequence, that's **H**ard, but this is all straightforward and a good high-level overview of the problem space.

**Pull request merged**